src/ofp_uipc_socket.c

```diff
@@ -222,9 +222,17 @@ void ofp_accept_unlock(void)
 	odp_rwlock_write_unlock(&shm->ofp_accept_mtx);
 }
 
+static uint64_t ofp_socket_get_shm_size(void)
+{
+	return sizeof(*shm) +
+		global_param->socket.num_max * sizeof(struct socket) +
+		global_param->socket.num_max * sizeof(struct sleeper);
+}
+
 static int ofp_socket_alloc_shared_memory(void)
 {
-	shm = ofp_shared_memory_alloc(SHM_NAME_SOCKET, sizeof(*shm));
+	shm = ofp_shared_memory_alloc(SHM_NAME_SOCKET,
```

Comment:
I think the code could be cleaner if socket_list and sleeper_list had simply separate SHM allocations instead of combined with that of ofp_socket_mem.